cdap-client/src/main/java/co/cask/cdap/client/MetadataClient.java

```diff
@@ -80,10 +81,28 @@ public MetadataClient(ClientConfig config) {
   public Set<MetadataSearchResultRecord> searchMetadata(Id.Namespace namespace, String query,
                                                         @Nullable MetadataSearchTargetType target)
     throws IOException, UnauthenticatedException {
+    Set<MetadataSearchTargetType> targets = ImmutableSet.of();
+    if (target != null) {
+      targets = ImmutableSet.of(target);
+    }
+    return searchMetadata(namespace, query, targets);
+  }
+
+  /**
+   * Returns search results for metadata.
+   *
+   * @param namespace the namespace to search in
+   * @param query the query string with which to search
+   * @param targets {@link MetadataSearchTargetType}s to search. If empty, all possible types will be searched
+   * @return A set of {@link MetadataSearchResultRecord} for the given query.
+   */
+  public Set<MetadataSearchResultRecord> searchMetadata(Id.Namespace namespace, String query,
+                                                        Set<MetadataSearchTargetType> targets)
+    throws IOException, UnauthenticatedException {
 
     String path = String.format("metadata/search?query=%s", query);
-    if (target != null) {
-      path = path + "&target=" + target;
+    for (MetadataSearchTargetType t : targets) {
+      path += "&target=" + t;
```

Comment:
I tried using `StringBuilder`. Found that it was too verbose without adding any benefit in this case.